### ISSUE TITLE
Dead/Incorrect Keys with identical Keyboard layouts

### ISSUE THREAD

**Issue by Xaymar** [input-leap/input-leap#1455]
### What happened?

On certain keyboard layouts, Input Leap fails to send key combinations to the client machines correctly. The reason is unknown, but it has been reported numerous times before (see Additional Information). On a German (QWERTZ) layout, the following all do not work correctly

- Press Ctrl + Print Screen
  **Expected:** Ctrl + Print Screen
  **Actual:** Ctrl
- Press Ctrl + Shift + Print Screen
  **Expected:** Ctrl + Shift + Print Screen
  **Actual:** Nothing
- Press Shift and ^.
  **Expected:** Press Shift and ^, resulting in °
  **Actual:** Nothing
- Press Shift and ´.
  **Expected:** Press Shift and ´, resulting in \`
  **Actual:** Nothing
- Press Scroll Lock
  **Expected:** Scroll Lock
  **Actual:** Nothing

This issue is unique to "synergy-core" based KVM software, and can usually only be observed when the Server has a keyboard layout different from "English (QWERTY)" on Windows. It appears that the software is for some reason converting keystrokes into something else entirely, instead of simply sending the keystrokes over to the other machine like a KVM switch would do.

I found that Microsofts Mouse Without Borders does not suffer from this problem, however it suffers from a lot of other problems that I consider much worse.

### Version

From Git HEAD or commit (specify below)

### Git commit hash (if applicable)

653e4bad

### If applicable, where did you install Barrier from?

https://dev.azure.com/debauchee/Barrier/_build/results?buildId=770&view=results

### What OSes are you seeing the problem on? (Check all that apply)

Windows

### What OS versions are you using?

Microsoft Windows Version 21H2 (OS Build 19044.1566)
Microsoft Windows Version 21H2 (OS Build 19044.1526)

### Relevant log output

```shell
For random key combinations, these log lines are spammed into the log:
[2022-03-02T06:16:56] INFO: found key in group 2
[2022-03-02T06:17:01] INFO: found key in group 2
```

### Any other information

Previous Reports of similar issues: #1307 #1267 #1230 #1229 #1190 #1186 #1181 #1143 #1119 #1082 #1041 #985 #975 #964 #943 #776 #727 #630 #619 #603 #576 #532 #512 #371 #136 #100 #45

Unrelated but also observed: #1109 #1061

There may be a lot more keys and combinations that are affected by this oddity.

**Comment by Xaymar** [comment]
After a bit of testing, I found additional behavior that changes depending on if the keyboard layout is anything except en-US QWERTY:

- Mouse3 to Mouse5 lose Press/Release support and only support individual Clicks.
- Modifier Keys (Ctrl, Shift, Alt, Capslock, AltGr, Menu) sometimes get stuck virtually, resulting in future key presses incorrectly combining with released modifier keys.
- Media Keys are not filtered correctly and run on both the target and the host machine, or are completely intercepted.
- Scroll Lock does not appear to work at all.
- Pause does not appear to work at all.

**Comment by ruggertech** [comment]
I'm experiencing an issue of not being able to write in Hebrew on the client machine when I use a en-US layout on the server machine.  However, when I set the server layout to HE, both layouts on the client work.  

I installed spanish ISO on both machines - the server machine allows all letters including ñ for example.
When moving the cursor to the client machine - Spanish special characters are not working well.
The ñ letter is translated to a "n" with space after it.  I tried it in Mac Notes on both machines.

### If applicable, where did you install Barrier from?
brew install barrier

### What OSes are you seeing the problem on? (Check all that apply)
MacOS

### What OS versions are you using?
Server:  MacOS Monterey 12.2.1, Macbook pro 2020
Client:  MacOS Monterey 12.3, Macbook pro Mid 2015

**Comment by Marenz** [comment]
Possibly related: https://github.com/input-leap/input-leap/issues/186

**Comment by meirtolpin11** [comment]
any solution for this? 
I tried to set Hebrew on my Windows (server side) but when trying to type on MacOS (client) I get question marks both for hebrew and english... 
if I set Server language to English, on client english works fine but Hebrew doesn't show anything

**Comment by shymega** [comment]
> any solution for this? I tried to set Hebrew on my Windows (server side) but when trying to type on MacOS (client) I get question marks both for hebrew and english... if I set Server language to English, on client english works fine but Hebrew doesn't show anything

No, and if there's no update here, or in other issues, *please* don't bump the issue, as it doesn't add anything to the discussion. We're aware of the problem, but we don't have the manpower to fix it. Language support isn't easy. Thank you. I will hide your comment now to tidy up the discussion.
